pkg/accelerator/accelDeviceProvider.go

```diff
@@ -25,22 +25,30 @@ import (
 )
 
 type accelDeviceProvider struct {
-	deviceList []types.AccelDevice
+	deviceList []*ghw.PCIDevice
 	rFactory   types.ResourceFactory
 }
 
 // NewAccelDeviceProvider DeviceProvider implementation from accelDeviceProvider instance
 func NewAccelDeviceProvider(rf types.ResourceFactory) types.DeviceProvider {
 	return &accelDeviceProvider{
 		rFactory:   rf,
-		deviceList: make([]types.AccelDevice, 0),
+		deviceList: make([]*ghw.PCIDevice, 0),
 	}
 }
 
-func (ap *accelDeviceProvider) GetDevices() []types.PciDevice {
-	newPciDevices := make([]types.PciDevice, len(ap.deviceList))
-	for i := range ap.deviceList {
-		newPciDevices[i] = ap.deviceList[i]
+func (ap *accelDeviceProvider) GetDiscoveredDevices() []*ghw.PCIDevice {
+	return ap.deviceList
+}
+
+func (ap *accelDeviceProvider) GetDevices(rc *types.ResourceConfig) []types.PciDevice {
+	newPciDevices := make([]types.PciDevice, 0)
+	for _, device := range ap.deviceList {
+		if newDevice, err := NewAccelDevice(device, ap.rFactory); err == nil {
```

Comment:
This doesn't add ResourceConfig for NewAccelDevice. Is it because accelerator device doesn't have special flag or it is missed?

Reply:
It's because it doesn't have any special flag. Since there is no interface that the device "constructors" have to follow, it can be `deviceType`-dependent.